### PR TITLE
:tada: Implement functionality to preview layer blend modes

### PR DESCRIPTION
**Pull request by niwinz**
Replaces the MR: https://github.com/penpot/penpot/pull/3211

### CODE REVIEW

File: frontend/src/app/main/refs.cljs [hunk 1]
```diff
@@ -550,3 +550,8 @@
 (def current-file-id
   (l/derived :current-file-id st/state))
 
+(def preview-blend-modes
+  (l/derived :preview-blend-modes workspace-local))
+
+(defn preview-blend-modes-by-id [id]
+  (l/derived #(get % id) preview-blend-modes =))
```
Comment (niwinz) [inline frontend/src/app/main/refs.cljs]: instead of `#(get % id)`, it should be `(l/key id)`

File: frontend/src/app/main/data/workspace.cljs [hunk 2]
```diff
@@ -2159,6 +2159,24 @@
...
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Preview blend modes
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn set-preview-blend-mode
+  [ids blend-mode]
+  (ptk/reify ::set-preview-blend-mode
+    ptk/UpdateEvent
+    (update [_ state]
+      (reduce #(assoc-in %1 [:workspace-local :preview-blend-modes %2] blend-mode) state ids))))
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace.cljs]: I guess they don't need to be on `:workspace-local`, prefer using a single level... some thing like `:workspace-prevew-blend` as a root key on state

File: frontend/src/app/main/ui/shapes/shape.cljs [hunk 3]
```diff
@@ -54,25 +55,30 @@
         pointer-events   (unchecked-get props "pointer-events")
         disable-shadows? (unchecked-get props "disable-shadows?")
 
+        preview-blend-mode-ref
+        (mf/use-memo (mf/deps (:id shape)) #(refs/preview-blend-modes-by-id (:id shape)))
```
Comment (niwinz) [inline frontend/src/app/main/ui/shapes/shape.cljs]: - prefer use `with-memo` instead use-memo,
- extract `shape-id` as separate binding on let and then use it (for avoid calling `(:id shape)`  twice
- replace all usage of `(:id shape)` on the rest of the shape, the `(:id shape)` is a Trie Lookup operation and for performance reasons is better lookup once and then reuse

**Comment by superalex** [comment]
Hello @akshay-gupta7 

After your commit we have added two more:
 - @niwinz did a refactor related to performance
 - I added a commit to fix some problems with selection multiple and to avoid savings while just doing the preview of the blend mode

**Pull request merged**